### DISCUSSION TITLE
Allow new project to API token

### DISCUSSION THREAD

**Discussion by samalrandula** [Unleash/unleash#2458] [Q&A] [answered]
I have created API token with access to feature toggles in project-a and proxy configured with that API token. Then I created project-b and several feature toggles inside project-b. I want to to allow project-b feature toggles through the proxy without restarting the proxy or configuring new proxy. Is there any way to do this? appreciate if you can help me soon

**Comment by thomasheartman** [comment, marked as answer]
Hey, @samalrandula 👋🏼 

Depending on how you have configured your token, that may be possible. When you create API tokens, you can choose to give access to a specific set of projects, or all (current and future) projects, as mentioned in the [format-section of the API token docs](https://docs.getunleash.io/reference/api-tokens-and-client-keys#format).

Thus, if your token was configured for all projects, it should just work. If it was configured only for project-a specifically, then it's not possible at the moment, I'm afraid. Once API tokens have been created, they can not be changed.

**Reply by samalrandula** [reply]
Thank you for the reply. I have created the API token only for project-a (say token-a). Therefore proxy allow only project-a and don't allow project-b at the moment. Now I have created API token specifically for project-b ( say token-b)  and another API token specifically for project-a and project-b ( say token-ab). Can I add the token-b to the proxy or replace token-a by token-ab without restarting the proxy or configuring new proxy?

**Reply by thomasheartman** [reply]
Got it; thanks for the clarification! Unfortunately, the proxy does not support updating keys at runtime, so you'd have to start a new proxy for the change to take effect.
